### ISSUE TITLE
wildcard "*" not fire the callbacks and "(*)" throw exception

### ISSUE THREAD

**Issue by yonathan9669** [visionmedia/page.js#307]
Hi, i'm building a polymer app using page as middleware for the routing but i'm getting stuck with some issues.

Quick intro: I separate the routing structure on 5 files, in which 3 of them has the functions definitions for the callbacks corresponding to what i need for each one of them. 1 file correspond to the share callbacks that the last 3 will require. The last file is where i setup the routing, i mean where i define when to call what.

Here is the problem:
Using the wildcard entry should load/save the user information from the context.state object, and manage the parameters before the url could get fully handled. but it's NEVER been called the loadUser function.
```
...
//Initial loads
page('/account/*', app.loadUser);
//page('/account/:organizationId/:pageId(*)', app.loadParams);
...
```
In the entry with the named wildcard throw an exception just for be there, i mean the app can't even load if it's uncomment.
![Exception capture](https://dl.dropboxusercontent.com/u/7006756/Reports/Github/pagejs/exception.png)

This work great, all the functions are called as it should, but obviously they don't work as it suppose to be, because the loadUser function isn't been called.
```
...
//Main redirection
page('/account/:organizationId/:pageId', app.breadcrumbs.mainPage, app.sidebar.mainPage, app.pages.mainPage);
page('/account/:organizationId/:pageId/*/new', app.breadcrumbs.mainPage, app.sidebar.newPage, app.pages.newPage);
...
```
This entry is not working, but i'm not entirely sure that you support this kind of expression, i suppose you do.

Here is the same thing, doesn't work just because a named wildcard it's been using
```
...
//Final loads
//page('/account/:organizationId/:pageId(*)', app.loadOrg);
...
```

This definition it's actually working as it should.
```
...
//404 NOT FOUND management
page('*', app.routes.notfound);
...
```

I thought i was doing something wrong and start to search on the web for this behavior, when i didn't find out anything, i start to make some tests with the EXAMPLES THAT YOU PROVIDE, and specifically in the profile example at the app.js file says:
```
...
// display the index page again after 5s
// only for user related pages
page('/user/*', displayIndexAfter(5000));
...
```
And guest what, this isn't working either.

So i really believe you have a bug with the entries that use wildcards

**Comment by JChapman202** [comment]
This is a bug in the page.js (there are quite a few bugs in the docs from what I've seen).  In order for the wildcards to work you must enter them as the following:

```
page('/user/(.*)', displayIndexAfter(5000));
```

Keep in mind it's really just describing a regular expression and (*) isn't a valid regular expression but (.*) would be a valid expression where it matches any number of any character.  The reason '*' works is because they actually replace '*' with '(.*)' when it's provided on it's own.

**Comment by gertsonderby** [comment]
Page.js (mainline) currently uses an old, somewhat broken verison of pathToRegexp (the library that Page.js and Express use to convert the path definitions to regexes that can be matched). The current version of that library is v1.2.1. I solved that by creating a new npm-package with my own changes. I advise you do that same. Or work around it as described above.

**Comment by woodlandhunter** [comment]
@yonathan9669 Could you test the current `master` branch and confirm it's now working? 

The path-to-regexp dependency has been updated which should resolve this issue.

**Comment by woodlandhunter** [comment]
@shuvalov-anton Recommend closing. Fixed with 1.6.4 release.
